src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,195 @@
+            case "delete":
+                return new DeleteCommand(Integer.parseInt(segments[1].trim()));
+            case "find":
+                if (segments[1].length() == 0) {
+                    throw new IndexOutOfBoundsException();
+                }
+                return new FindCommand(segments[1].trim().split(" "));
+            case "update":
+                String[] updateSegments = segments[1].split(" ", 2);
+                if (updateSegments[1].trim().length() == 0) {
+                    throw new DukeException("Empty description is not valid");
+                }
+                return new UpdateCommand(
+                        Integer.parseInt(updateSegments[0].trim()),
+                        updateSegments[1].trim());
+            case "todo":
+            case "deadline":
+            case "event":
+                break;
+            default:
+                throw new DukeException("Aww! The first word of your input is wrong!");
+            }
+        } catch (NumberFormatException | IndexOutOfBoundsException e) {
+            throw new DukeException("Please check your inputs again, ensure words are spaced and numbers"
+                    + "(if any) are correct.");
+        }
+
+        String[] parsedSegments = parseSegments(input, cmd, getTaskBreakPt(cmd));
+        // CHECK FOR TASKS-RELATED COMMANDS
+        switch (cmd) {
+        case "todo":
+            return new TodoCommand(parsedSegments[0]);
+        case "deadline":
+            return new DeadlineCommand(parsedSegments[0], parsedSegments[1]);
+        case "event":
+            return new EventCommand(parsedSegments[0], parsedSegments[1]);
+        default:
+            throw new DukeException("Aww! The first word of your input is wrong!");
+        }
+    }
+
+    /**
+     * Parses raw data that is loaded from save file.
+     *
+     * @param data The raw data.
+     * @return The correct Task that corresponds to what the raw data represents.
+     * @throws DukeException If data is unexpected or exceptional.
+     */
+    public static Task parseTaskData(String data) throws DukeException {
+        String[] segments = data.split("\\|");
+        Task task;
+        boolean isDone = segments[1].equals("1");
+
+        try {
+            switch (segments[0]) {
+            case "T":
+                task = new TodoTask(segments[2], isDone);
+                break;
+            case "D":
+                task = new DeadlineTask(segments[2], isDone, LocalDate.parse(segments[3]));
+                break;
+            case "E":
+                task = new EventTask(segments[2], isDone, LocalDate.parse(segments[3]));
+                break;
+            default:
+                task = null;
+            }
+            return task;
+        } catch (DateTimeParseException e) {
+            throw new DukeException("supplied data: " + segments[3] + " does not conform to yyyy-mm-dd");
+        }
+
+    }
```

Comment:
You can consider handling the conversion of the text data to the respective tasks in the Storage class, as the main purpose of the parser is to only make sense of the user input.